pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -71,19 +86,26 @@ def render(self, task) -> Text:
                 metrics = self._trainer.progress_bar_callback.get_metrics(self._trainer, self._pl_module)
             else:
                 metrics = self._trainer.progress_bar_metrics
+
             for k, v in metrics.items():
                 _text += f"{k}: {round(v, 3) if isinstance(v, float) else v} "
             text = Text.from_markup(_text, style=None, justify="left")
             return text
 
 
-STYLES: Dict[str, str] = {
-    "train": "red",
-    "sanity_check": "yellow",
-    "validate": "yellow",
-    "test": "yellow",
-    "predict": "yellow",
-}
+@dataclass
+class RichProgressBarTheme:
+    """Styles to associate to different base components.
+
+    https://rich.readthedocs.io/en/stable/style.html
+    """
+
+    text_color: str = "white"
+    progress_bar_complete: Union[str, Style] = "#6206E0"
+    progress_bar_finished: Union[str, Style] = "#6206E0"
+    batch_process: str = "white"
+    time: str = "grey54"
+    processing_speed: str = "grey70"
```

Comment:
@SeanNaren @kaushikb11 

noob question: Why did we introduce this new class `RichProgressBarTheme` and pass an instance to `RichProgressBar`? I'm asking because if we didn't have this class and passed colours (or a dictionary of colours) directly to `RichProgressBar`, colour customisation would be easier for users in my opinion. (One less thing to remember :])

```diff
from pytorch_lightning.callbacks import RichProgressBar
-from pytorch_lightning.callbacks.progress.rich_progress import RichProgressBarTheme

progress_bar = RichProgressBar(
-    theme=RichProgressBarTheme(
        description="green_yellow",
        progress_bar="green1",
        progress_bar_finished="green1",
        batch_progress="green_yellow",
        time="grey82",
        processing_speed="grey82",
        metrics="grey82",
-    )
)
```

FYI, I'm looking at the example in the blog post: https://devblog.pytorchlightning.ai/super-charged-progress-bars-with-rich-lightning-669653d6ab97